### PR TITLE
Refactor(eos_cli_config_gen): Wildcard dict to list for vrfs

### PR DESCRIPTION
**Pull request by ghost**
## Data model key

`vrfs`

### Contributor Checklist

- [x] Update `README_v4.0.md` with new data model
- [x] Update all `host_vars` under molecule scenario `eos_cli_config_gen_v4.0` with new data model
- [x] Update `templates/eos/< template >.j2` and `templates/documentation/< template >.j2`:
  - [x] Add `arista.avd.convert_dicts('<primary key>')` filter for loops previously using wildcard keys
  - [x] Update `arista.avd.natural_sort('<primary key>')` to sort on the primary key (if applicable)
- [x] Run molecule `cd ansible_collections/arista/avd/molecule ; make cli-4.0-schema`
  - [x] Verify no changes to generated configs/docs

### Reviewer Checklist

- Reviewer 1: Tony
  - [x] Verify data model changes
  - [x] Verify that all molecule `host_vars` under `eos_cli_config_gen_v4.0` has been updated to the new data model
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

- Reviewer 2: Claus
  - [x] Verify data model changes
  - [x] Verify that all molecule `host_vars` under `eos_cli_config_gen_v4.0` has been updated to the new data model
  - [x] Verify no changes to configs/docs on any molecule scenario
  - [x] Verify that CI pass

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/vrf-instances.j2 [hunk 1]
```diff
@@ -1,18 +1,18 @@
...
 # VRF Instances
 
 ## VRF Instances Summary
 
 | VRF Name | IP Routing |
 | -------- | ---------- |
-{%     for vrf in vrfs | arista.avd.natural_sort %}
-{%         if vrfs[vrf].ip_routing == true %}
+{%     for vrf in vrfs | arista.avd.convert_dicts('name') | arista.avd.natural_sort('name') %}
+{%         if vrf.ip_routing == true %}
```
Comment (tgodaA) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/vrf-instances.j2]: This has to be **vrf.ip_routing is arista.avd.defined(true)**

Comment (tgodaA) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/documentation/vrf-instances.j2]: Looks like the previous bug.

**Review by tgodaA — APPROVED** [review]
LGTM

**Review by ClausHolbechArista — APPROVED** [review]
LGTM

**Pull request merged**